### PR TITLE
Keyvault changelog

### PR DESCRIPTION
**Pull request by vhvb1989**

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-common/vcpkg/CONTROL [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+#
 Source: azure-security-keyvault-common-cpp
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-common/vcpkg/CONTROL:4]: @antkmsft, @danieljurek will these comments be removed as part of the vcpkg publish step just like for the other packages like core?
https://github.com/Azure/azure-sdk-for-cpp/blob/46c84e3a3f1bc6c6f32c9892e94324a05009b232/sdk/core/azure-core/vcpkg/CONTROL#L1-L4

https://github.com/microsoft/vcpkg/blob/9d807041270262cc108b8deda60550a2cd95db1a/ports/azure-core-cpp/CONTROL#L1-L2

Comment (danieljurek) [reply sdk/keyvault/azure-security-keyvault-common/vcpkg/CONTROL:4]: @antkmsft should correct me if I'm wrong -- 

If the CMakeLists.txt file for the library is using `az_vcpkg_export` it should remove that section from the file: https://github.com/Azure/azure-sdk-for-cpp/blob/master/cmake-modules/AzureVcpkg.cmake#L64

This should also be verifiable in the package artifact outputs: https://dev.azure.com/azure-sdk/public/_build/results?buildId=830651&view=artifacts&pathAsName=false&type=publishedArtifacts 

If the comment removal script is working properly those lines will not be in the outputs.

File: sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
 
 ### New Features
 
-- KeyVault client with the next operations:
-  - GetKey.
-  - CreateKey.
+- KeyVault Keys client for get, create, list, delete, backup, restore, import operations.
 - General purpose header `key_vault.hpp`.
-- KeyVault Keys types.
+- KeyVault Keys models.
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md:9]:
```suggestion
- Added model types which are returned from the `KeyClient` operations, such as `Azure::Security::KeyVault::Keys::KeyVaultKey`.
```

File: sdk/keyvault/azure-security-keyvault-common/vcpkg/CONTROL (hunk 1 above)
Comment (ahsonkhan) [reply sdk/keyvault/azure-security-keyvault-common/vcpkg/CONTROL:4]: > If the CMakeLists.txt file for the library is using `az_vcpkg_export` 

Looks like it is:
https://github.com/Azure/azure-sdk-for-cpp/blob/308e5363adaa00ace2f06ff0f1a31747a0a28062/sdk/keyvault/azure-security-keyvault-keys/CMakeLists.txt#L105-L109

Verified in the artifacts, the script is working :)

File: sdk/keyvault/azure-security-keyvault-common/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 4.0.0-beta.1 (Unreleased)
+## 4.0.0-beta.1 (2021-04-07)
```
[the remark this answers is not in the document]
Comment (ahsonkhan) [reply sdk/keyvault/azure-security-keyvault-common/CHANGELOG.md:3]: Update this CL:
```md
- Added `Azure::Security::KeyVault::KeyVaultException`.
```

File: sdk/keyvault/azure-security-keyvault-common/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
 
 ### New Features
```
[the remark this answers is not in the document]
Comment (vhvb1989) [reply sdk/keyvault/azure-security-keyvault-common/CHANGELOG.md:5]: We need it, yes. It has the KeyVaultPipeline component used by the client to send KeyVault requests.

File: sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md [hunk 5]
```diff
@@ -1,11 +1,10 @@
...
-## 4.0.0-beta.1 (Unreleased)
+## 4.0.0-beta.1 (2021-04-07)
 
 ### New Features
 
-- KeyVault client with the next operations:
-  - GetKey.
-  - CreateKey.
+- KeyVault Keys client for get, create, list, delete, backup, restore, import operations.
 - General purpose header `key_vault.hpp`.
```
[the remark this answers is not in the document]
Comment (vhvb1989) [reply sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md:8]: key vault is always 2 words. That's how is should be seen everywhere
key_vault  -> files
KeyVault   -> Class
keyVault   ->  Member

But, anyway, it should be key_vault_keys.hpp .  I will update it for preview 2

[the remark this answers is not in the document]
Comment (ahsonkhan) [reply sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md]: We should then be consistent everywhere.

**For example:**
`keyvault_constants`
`keyvault_pipeline`

https://github.com/Azure/azure-sdk-for-cpp/blob/46c84e3a3f1bc6c6f32c9892e94324a05009b232/sdk/keyvault/azure-security-keyvault-common/src/keyvault_pipeline.cpp

image

Would we name the folder as `key_vault` too?

How do the other source-distributed languages name this?

File: sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md [hunk 6]
```diff
@@ -1,11 +1,9 @@
...
+## 4.0.0-beta.1 (2021-04-07)
 
 ### New Features
 
-- KeyVault client with the next operations:
-  - GetKey.
-  - CreateKey.
-- General purpose header `key_vault.hpp`.
-- KeyVault Keys types.
+- Added `Azure::Security::KeyVault::Keys::KeyClient` for get, create, list, delete, backup, restore, and import operations.
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md:7]:
```suggestion
- Added `Azure::Security::KeyVault::Keys::KeyClient` for get, create, list, delete, backup, restore, and import key operations.
```

**Review by ahsonkhan — APPROVED** [review]
Other than small nit in `azure-security-keyvault-keys` CL and suggestion to enhance the `azure-security-keyvault-common/CHANGELOG.md` to use the fully qualified type name, looks good.

File: sdk/keyvault/azure-security-keyvault-common/CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (vhvb1989) [reply sdk/keyvault/azure-security-keyvault-common/CHANGELOG.md:3]: I don't want to advertise it

[the remark this answers is not in the document]
Comment (ahsonkhan) [reply sdk/keyvault/azure-security-keyvault-common/CHANGELOG.md:3]: But you already are.
https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/keyvault/azure-security-keyvault-common/CHANGELOG.md#new-features
image

File: sdk/keyvault/ci.yml [hunk 7]
```diff
@@ -39,5 +39,7 @@ stages:
       Artifacts:
         - Name: azure-security-keyvault-common
           Path: azure-security-keyvault-common
+          VcpkgPortName: azure-security-keyvault-common-cpp
```
Comment (ahsonkhan) [inline sdk/keyvault/ci.yml:42]: .NET calls this Core. Did we already discuss C++ sticking to `common`?
https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/keyvault/Microsoft.Azure.KeyVault.Core

**Pull request merged**